### PR TITLE
Filter out to single tick per interval.

### PR DESCRIPTION
**Pull request by vincentwschau**
### Changelist
It's possible for a vault to be instantiated in the middle of the day/hour, and start having PnL ticks in the middle of the day/hour. This breaks the aggregation logic for PnL ticks as it introduces an extra PnL tick in the middle of the day / hour for the megavault.
This change adds a filtering function that only takes the aggregated PnL ticks with block times closest to the start of the day / hour.

### Test Plan
Unit tests.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new configuration parameter for vault profit and loss history, allowing for more granular control.
	- Enhanced functionality for fetching historical profit and loss data with improved resolution handling and filtering logic.

- **Bug Fixes**
	- Updated test cases to ensure accurate handling of historical profit and loss data for vault subaccounts, accommodating multiple vaults and undefined values.

### CODE REVIEW

File: indexer/services/comlink/src/controllers/api/v4/vault-controller.ts [hunk 1]
```diff
@@ -294,21 +295,22 @@ router.get(
...
 ): Promise<PnlTicksFromDatabase[]> {
   if (vaultSubaccountIds.length === 0) {
     return [];
   }
-  let pnlTickInterval: PnlTickInterval;
-  if (resolution === undefined) {
-    pnlTickInterval = PnlTickInterval.day;
+
+  let windowSeconds: number;
+  if (resolution === PnlTickInterval.day) {
```
Comment (tqin7) [inline indexer/services/comlink/src/controllers/api/v4/vault-controller.ts:305]: nit: maybe define a function that converts from resolution to seconds? thinking about the possibility of adding more resolution values in the future

File: indexer/services/comlink/src/controllers/api/v4/vault-controller.ts [hunk 2]
```diff
@@ -470,6 +472,55 @@ function getPnlTicksWithCurrentTick(
...
+  // Track start of days to closest block by block time.
+  const blocksPerInterval: Map<string, number> = new Map();
+  // Track start of days to closest Pnl tick.
+  const ticksPerInterval: Map<string, PnlTicksFromDatabase> = new Map();
+  pnlTicksByBlock.forEach((pnlTick: PnlTicksFromDatabase, block: number): void => {
+    const blockTime: DateTime = DateTime.fromISO(pnlTick.blockTime).toUTC();
+    blockToBlockTime.set(block, blockTime);
+
+    const startOfInterval: DateTime = blockTime.toUTC().startOf(resolution);
+    const startOfIntervalBlock: number | undefined = blocksPerInterval.get(startOfInterval.toISO());
```
Comment (tqin7) [inline indexer/services/comlink/src/controllers/api/v4/vault-controller.ts:495]: nit: not sure how much work this call is. if notable, define `startOfInterval.toISO()` as a const so we don't call it many times?

**Review by tqin7 — APPROVED** [review]

File: indexer/services/comlink/src/controllers/api/v4/vault-controller.ts (hunk 1 above)
Comment (vincentwschau) [reply indexer/services/comlink/src/controllers/api/v4/vault-controller.ts:305]: This doesn't work with the underlying SQL query.

**Comment by vincentwschau** [comment]
>how are pnl ticks handled with generic subaccounts (what happens if a subaccount gets transferred to for the first time in the middle of a day/hour)?

A pnl tick only gets created at each hour.

**Review by tqin7 — APPROVED** [review]

**Comment by vincentwschau** [comment]
@Mergifyio backport release/indexer/v7.x

**Pull request merged**